be/src/service/internal_service.cpp

```diff
@@ -54,6 +54,18 @@ void PInternalServiceImpl<T>::transmit_data(google::protobuf::RpcController* cnt
             request->be_number(), request->packet_seq(),
             eos ? nullptr : &done);
     }
+
+    if (request->has_query_statistic()) {
```

Comment:
you should update query statistics in `add_data` function

Reply:
Now add_data is used for add batch,  when eos is true and request does't contain batch,  add_data will not be called, but request may contains query statistics, so it is more simple to handle query statistics update separately.